include/deal.II/multigrid/mg_transfer_global_coarsening.h

```diff
@@ -122,7 +127,9 @@ namespace MGTransferGlobalCoarseningTools
   std::vector<std::shared_ptr<const Triangulation<dim, spacedim>>>
   create_geometric_coarsening_sequence(
     const Triangulation<dim, spacedim> &                  tria,
```

Comment:
To be self-documenting, we should mark `tria` as non-const reference, as it might get changed.
```suggestion
    Triangulation<dim, spacedim> &                  tria,
```
However, that might force some user codes to `const_cast` even if they do not want to modify the triangulation. An alternative would be that we provide two variants of this function with different names, and only the second one with full flexibility. We could then redirect the first function to the second with right parameters (and `const_cast` in that call, but we know that nothing will happen).

Reply:
OK. I have created a new function.